src/genkey/clu_genkey.c

```diff
@@ -799,6 +799,209 @@ int wolfCLU_genKey_RSA(WC_RNG* rng, char* fName, int directive, int fmt, int
 #endif
 }
 
+
+int wolfCLU_genKey_Dilithium(WC_RNG* rng, char* fName, int directive, int fmt,
+                            int keySz, int level, int withAlg, int keyType)
+{
+#ifdef HAVE_DILITHIUM
+    dilithium_key key;
+    XFILE  file;
+    int    ret = WOLFCLU_SUCCESS;
+
+    int   fNameSz;
+    int   fExtSz      = 6;
+    char  fExtPriv[6] = ".priv\0";
+    char  fExtPub[6]  = ".pub\0\0";
+    char* fOutNameBuf = NULL;
+
+    #ifdef NO_AES
+    /* use 16 bytes for AES block size */
+    size_t maxDerBufSz = 4 * keySz * 16;
+    #else
+    size_t maxDerBufSz = 4 * keySz * AES_BLOCK_SIZE;
+    #endif  /* NO_AES */
+    byte*  derBuf      = NULL;
+    byte*  pemBuf      = NULL;
+    byte*  outBuf      = NULL;
+    int    derBufSz    = -1;
+    int    pemBufSz    = 0;
+    int    outBufSz    = 0;
+
+    if (rng == NULL || fName == NULL) {
+        return BAD_FUNC_ARG;
+    }
+
+    /* initialize the dilithium key */
+    if (wc_dilithium_init(&key) != 0) {
+        return WOLFCLU_FAILURE;
+    }
+
+    /* set the level of the dilithium key */
+    if (wc_dilithium_set_level(&key, level) != 0) {
+        return WOLFCLU_FAILURE;
+    }
+
+    /* make the dilithium key */
+    if (wc_dilithium_make_key(&key, rng) != 0) {
+        return WOLFCLU_FAILURE;
+    }
+
+    /* set up the file name output buffer */
+    if (ret == WOLFCLU_SUCCESS) {
+        fNameSz     = (int)XSTRLEN(fName);
+        fOutNameBuf = (char*)XMALLOC(fNameSz + fExtSz, HEAP_HINT,
```

Comment:
Sanity checks on the size before malloc. If fNameSz + fExtSz wraps around int type, fNameSz for some reason is negative, possibly upper bounds to avoid an abuse but is trickier to make a great upper bounds value here.